src/ui/overlay.rs

```diff
@@ -333,3 +377,24 @@ fn update_overlay_position_based_on_location(
         }
     }
 }
+
+/// Round `duration` down to the nearest `increment` and format as "Xh Ym" or "Xm"
+fn format_rounded_duration(duration: Duration, increment: Duration) -> String {
```

Comment:
This is only ever used with one granularity, so I'd move this to `src/ui/util.rs` and/or make it `format_minutes`, to be consistent with `format_seconds`.